analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/InitializationVectorShouldBeRandomBase.cs

```diff
@@ -18,10 +18,103 @@
+        }
+        else if (operation.AsAssignment() is { } assignment)
+        {
+            return ProcessAssignmentToIVProperty(state, assignment)
+                   ?? state;
+        }
+        else if (operation.AsPropertyReference() is { } property
+                 && IsIVProperty(state, property)
+                 && state[property.Instance]?.Constraint<ByteCollectionConstraint>() is { } constraint)
+        {
+            return state.SetOperationConstraint(property, constraint);
+        }
+        else if (operation.AsInvocation() is { } invocation)
+        {
+            return ProcessStrongRandomGeneratorMethodInvocation(state, invocation)
+                   ?? ProcessIsGenerateInvocation(state, invocation)
+                   ?? ProcessCreateEncryptorMethodInvocation(state, context, invocation)
+                   ?? state;
+        }
+        else
+        {
+            return state;
+        }
+    }
+
+    private static ProgramState ProcessAssignmentToIVProperty(ProgramState state, IAssignmentOperationWrapper assignment) =>
+        assignment.Target?.AsPropertyReference() is { } property
+        && IsIVProperty(state, property)
+        && state[assignment.Value].HasConstraint(ByteCollectionConstraint.CryptographicallyWeak)
+            ? state.SetSymbolConstraint(state.ResolveCaptureAndUnwrapConversion(property.Instance).TrackedSymbol(), ByteCollectionConstraint.CryptographicallyWeak)
+            : null;
+
+    private static ProgramState ProcessStrongRandomGeneratorMethodInvocation(ProgramState state, IInvocationOperationWrapper invocation)
+    {
+        if (IsCryptographicallyStrongRandomNumberGenerator(invocation)
+            && invocation.ArgumentValue("data") is { } byteArray
+            && state.ResolveCaptureAndUnwrapConversion(byteArray).TrackedSymbol() is { } byteArraySymbol)
+        {
+            return state.SetSymbolConstraint(byteArraySymbol, ByteCollectionConstraint.CryptographicallyStrong);
+        }
+        {
+            return null;
+        }
+    }
+
+    private static ProgramState ProcessIsGenerateInvocation(ProgramState state, IInvocationOperationWrapper invocation) =>
+        invocation.TargetMethod.Name.Equals(nameof(SymmetricAlgorithm.GenerateIV))
+        && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm)
+            ? state.SetSymbolConstraint(invocation.Instance.TrackedSymbol(), ByteCollectionConstraint.CryptographicallyStrong)
+            : null;
+
+    private ProgramState ProcessCreateEncryptorMethodInvocation(ProgramState state, SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        if (IsCreateEncryptorMethod(invocation)
+            && (ArgumentIsCryptographicallyWeak(state, invocation) || UsesCryptographicallyWeakIVProperty(state, invocation)))
+        {
+            ReportIssue(context.Operation.Instance, invocation.Instance.Syntax.ToString());
```

Comment:
You don't need context in this method.
```suggestion
            ReportIssue(invocation.WrappedOperation, invocation.Instance.Syntax.ToString());
```